### DISCUSSION TITLE
Use JavaPoet for AP?

### DISCUSSION THREAD

**Discussion by mechite** [avaje/avaje-inject#267] [General]
Right now from what I've seen, generated code is written by hand

**Comment by SentryMan** [comment]
Perhaps, but I personally like my libs to have as few dependencies as possible. The writer classes we have are organized pretty well (to me at least). Is there some functionality JavaPoet has that would further improve the class writing?

**Comment by mechite** [comment]
JavaPoet handles imports and whatnot automagically and is a very small, minimal library just like practically everything Square has made; it would just contribute to the readability & maintainability of avaje-inject as things would be much more concise and less boilerplate writer utilities to maintain

While I don't know a ton about codegen/annotation processing, I don't particularly like the fact that the generated code is full of `var` usages and the like.

**Reply by SentryMan** [reply]
var definitely helps because you don't need code to add an explicit import.

**Reply by mechite** [reply]
> var definitely helps because you don't need code to add an explicit import.

well thats the point, you can use javapoet for these things & it will save you the burden of adding other required imports
var just makes for more convoluted code

**Reply by SentryMan** [reply]
I mean in situations where you wouldn't have the import classes on hand. Or like when you don't want to deal with name conflicts.
